### PR TITLE
Docs: Improve and add new parallel routes examples

### PR DESCRIPTION
**Pull request by delbaoliveira**
Continuing from: https://github.com/vercel/next.js/pull/60806

  - [x] Update **conditional routes** example to render dashboard pages based on user's role (rather than checking if the user is logged in)
  - [x] Add **tab group** example to allow each slot to be navigated independently. 
  - [x] Expand the **modal** example to show full implementation
    - [x] Opening the modal
    - [x] Closing the modal
- [ ] Add **catch-all** slots example
    
Please merge after the diagrams PR: https://github.com/vercel/front/pull/28770

### CODE REVIEW

File: docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx [hunk 1]
```diff
@@ -130,125 +133,275 @@ When a user navigates to `app/@auth/login` (or `/login` in the URL bar), `loginS
...
+
+```jsx filename="app/@auth/login/page.js" switcher
+import { Login } from '@/app/ui/login'
+
+export default function Page() {
+  return <Login />
+}
+```
+
+Then, create a new `@modal` slot, and inside it, add a `default.js` file that returns `null`. This ensures that the modal is not rendered when it's not active.
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx:261]: Maybe worth linking out to the `default` docs here again

File: docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx [hunk 2]
```diff
@@ -130,125 +133,275 @@ When a user navigates to `app/@auth/login` (or `/login` in the URL bar), `loginS
...
     </Modal>
   )
 }
 ```
 
-To ensure that the contents of the modal don't get rendered when it's not active, you can create a `default.js` file that returns `null`.
+> **Good to know:**
+>
+> - The convention used to intercept the route, e.g. `(.)`, depends on your file-system structure. See [Intercepting Routes convention](/docs/app/building-your-application/routing/intercepting-routes#convention).
+> - By separating the `<Modal>` functionality from the modal content (`<Login>`), you can ensure any content inside the modal, e.g. [forms](/docs/app/building-your-application/data-fetching/server-actions-and-mutations#forms), are Server Components. See [interleaving Client and Server Components](/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props) for more information.
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx:306]:
```suggestion
> - By separating the `<Modal>` functionality from the modal content (`<Login>`), you can ensure any content inside the modal, e.g. [forms](/docs/app/building-your-application/data-fetching/server-actions-and-mutations#forms), are Server Components. See [Interleaving Client and Server Components](/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props) for more information.
```

File: docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx [hunk 3]
```diff
@@ -130,125 +133,275 @@ When a user navigates to `app/@auth/login` (or `/login` in the URL bar), `loginS
...
+import { Login } from '@/app/ui/login'
+
+export default function Page() {
+  return <Login />
+}
+```
+
+Then, create a new `@modal` slot, and inside it, add a `default.js` file that returns `null`. This ensures that the modal is not rendered when it's not active.
+
+```tsx filename="app/@modal/default.tsx"
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx:263]:
```suggestion
```tsx filename="app/@modal/default.tsx" switcher
```

File: docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx [hunk 4]
```diff
@@ -130,125 +133,275 @@ When a user navigates to `app/@auth/login` (or `/login` in the URL bar), `loginS
...
+export default function Default() {
+  return null
+}
+```
 
-export default function Login() {
+```tsx filename="app/@modal/default.tsx"
```
Comment (ztanner) [inline docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx:269]:
```suggestion
```tsx filename="app/@modal/default.js" switcher
```

**Comment by ztanner** [comment]
One use-case that might be worth documenting is catch-all slots. ie, you could have a sidebar that matches all sub-paths, rather than falling back to `default`, with `@sidebar/[...catchAll]/page.tsx`.

File: docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx [hunk 5]
```diff
@@ -71,17 +77,17 @@ However, slots are **not** [route segments](/docs/app/building-your-application/
...
-- **Hard Navigation**: After a full-page load (browser refresh), Next.js cannot determine the active state of slots that don't match the current URL. Instead, it will render a [`default.js`](#defaultjs) file for the unmatched slots, or `404` if `default.js` doesn't exist.
+- **Hard Navigation**: After a full-page load (browser refresh), Next.js cannot determine the active state for the slots that don't match the current URL. Instead, it will render a [`default.js`](#defaultjs) file for the unmatched slots, or `404` if `default.js` doesn't exist.
 
 > **Good to know**:
 >
-> - The `404` for unmatched routes helps ensure that you don't accidentally render a route that shouldn't be parallel rendered.
+> - The `404` for unmatched routes helps ensure that you don't accidentally render a route that is not a parallel.
```
[the remark this answers is not in the document]
Comment (delbaoliveira) [reply docs/02-app/01-building-your-application/01-routing/10-parallel-routes.mdx:84]: I was struggling with this one, this is better wording, thank  you!

**Comment by delbaoliveira** [comment]
Converting the PR back to draft as we’re currently clarifying behaviour for `[…catchAll]` and use cases for conditional routes.

**Review by leerob — APPROVED** [review]

**Pull request merged**